crates/bevy_ecs/src/schedule_v3/config.rs

```diff
@@ -198,12 +209,37 @@ impl IntoSystemSetConfig for SystemSetConfig {
             !set.is_system_type(),
             "adding arbitrary systems to a system type set is not allowed"
         );
+        assert!(
+            !set.is_base(),
+            "Sets cannot be added to 'base' system sets using 'in_set'. Use 'in_base_set' instead."
+        );
+        assert!(
+            !self.set.is_base(),
+            "Base system sets cannot be added to other sets."
+        );
         self.graph_info.sets.push(Box::new(set));
         self
     }
 
-    fn no_default_set(mut self) -> SystemSetConfig {
-        self.graph_info.add_default_set = false;
+    fn in_base_set(mut self, set: impl SystemSet) -> Self {
+        assert!(
+            !set.is_system_type(),
+            "adding arbitrary systems to a system type set is not allowed"
+        );
+        assert!(
+            set.is_base(),
+            "Normal sets cannot be added to system sets using 'in_base_set'. Use 'in_set' instead."
```

Comment:
This error message is confusing. It should be something more like "normal sets cannot be added to normal sets using in_base_sets."